contrib/mesos/pkg/election/etcd_master_test.go

```diff
@@ -57,77 +54,21 @@ func TestEtcdMasterNoOther(t *testing.T) {
 	w.Stop()
 }
 
-// MockClient is wrapper aroung tools.EtcdClient.
-type MockClient struct {
-	client tools.EtcdClient
-	t      *testing.T
-	// afterGetFunc is called after each Get() call.
-	afterGetFunc func()
-	calls        []string
-}
-
-func (m *MockClient) GetCluster() []string {
-	return m.client.GetCluster()
-}
-
-func (m *MockClient) Get(key string, sort, recursive bool) (*etcd.Response, error) {
-	m.calls = append(m.calls, "get")
-	defer m.afterGetFunc()
-	response, err := m.client.Get(key, sort, recursive)
-	return response, err
-}
-
-func (m *MockClient) Set(key, value string, ttl uint64) (*etcd.Response, error) {
-	return m.client.Set(key, value, ttl)
-}
-
-func (m *MockClient) Create(key, value string, ttl uint64) (*etcd.Response, error) {
-	m.calls = append(m.calls, "create")
-	return m.client.Create(key, value, ttl)
-}
-
-func (m *MockClient) CompareAndSwap(key, value string, ttl uint64, prevValue string, prevIndex uint64) (*etcd.Response, error) {
-	return m.client.CompareAndSwap(key, value, ttl, prevValue, prevIndex)
-}
-
-func (m *MockClient) Delete(key string, recursive bool) (*etcd.Response, error) {
-	return m.client.Delete(key, recursive)
-}
-
-func (m *MockClient) Watch(prefix string, waitIndex uint64, recursive bool, receiver chan *etcd.Response, stop chan bool) (*etcd.Response, error) {
-	return m.client.Watch(prefix, waitIndex, recursive, receiver, stop)
-}
-
 func TestEtcdMasterNoOtherThenConflict(t *testing.T) {
 	server := etcdtesting.NewEtcdTestClientServer(t)
 	defer server.Terminate(t)
 
-	// We set up the following scenario:
-	// - after each Get() call, we write "baz" to a path
-	// - this is simulating someone else writing a data
-	// - the value written by someone else is the new value
 	path := "foo"
-	client := &MockClient{
-		client: server.Client,
-		t:      t,
-		afterGetFunc: func() {
-			if _, err := server.Client.Set(path, "baz", 0); err != nil {
-				t.Errorf("unexpected error: %v", err)
-			}
-		},
-		calls: make([]string, 0),
-	}
+	master := NewEtcdMasterElector(server.Client)
+	leader := NewEtcdMasterElector(server.Client)
 
-	master := NewEtcdMasterElector(client)
+	w_ldr := leader.Elect(path, "baz")
+	result := <-w_ldr.ResultChan()
 	w := master.Elect(path, "bar")
-	result := <-w.ResultChan()
+	result = <-w.ResultChan()
 	if result.Type != watch.Modified || result.Object.(Master) != "baz" {
 		t.Errorf("unexpected event: %#v", result)
 	}
 	w.Stop()
-
-	expectedCalls := []string{"get", "create", "get"}
-	if !reflect.DeepEqual(client.calls, expectedCalls) {
-		t.Errorf("unexpected calls: %#v", client.calls)
-	}
+	w_ldr.Stop()
```

Comment:
I would argue no, b/c the goal of the test is to check that (X) is not the leader.  

Operational ordering imho was a false test that wrapped the tool.interface.